src/components/IFrame.js

```diff
@@ -97,38 +190,102 @@ const propTypes = {
+// If the top most central pane route is not an iframe route, then it returns undefined.
+function getIframeRouteNameAndParams(state) {
+    const routeName = getTopMostCentralPaneRouteName(state);
+    if (routeName !== undefined && routeName !== 'Report') {
+        const topMostCentralPane = lodashFindLast(state.routes, (route) => route.name === 'CentralPaneNavigator');
+        if (topMostCentralPane && topMostCentralPane.state) {
+            const params = topMostCentralPane.state.routes[topMostCentralPane.state.index || 0].params;
+            return {routeName, params, index: state.index};
+        }
 
-    useEffect(() => {
-        setOldDotURL(`https://expensify.com.dev/${getOldDotURL(window.location.href)}`);
+        const params = topMostCentralPane.params;
+        if (params) {
+            return {routeName, params: params.params, index: state.index};
+        }
+    }
+    return {routeName: undefined, params: undefined, index: state.index};
+}
+
+// TODO: use proper URL
+const BASE_IFRAME_URL = 'http://localhost:3000';
+
+function IFrame({session}) {
+    const [oldDotURL, setOldDotURL] = useState(undefined);
+    const {routeName, params, index} = useNavigationState((state) => getIframeRouteNameAndParams(state));
 
+    useEffect(() => {
         window.addEventListener('message', (event) => {
-            const url = event.data;
+            let data = {};
+            try {
+                data = JSON.parse(event.data);
+            } catch {
+                data = {};
+            }
+
+            if (data.iframeId !== 'OldDot' || data.type !== 'URL_CHANGED') {
+                return;
+            }
+
+            const {url} = data.payload ?? {};
+
             // TODO: use this value to navigate to a new path
             // eslint-disable-next-line no-unused-vars
             const newDotURL = getNewDotURL(url);
+
+            // TODO: The line below should be enough but I can't test it because of cookies issue.
+            // navigation.navigate(newDotURL);
         });
     }, []);
 
     useEffect(() => {
-        document.cookie = `authToken=${session.authToken}; domain=expensify.com.dev; path=/;`;
-        document.cookie = `email=${session.email}; domain=expensify.com.dev; path=/;`;
+        // The routeName will be undefined if there is no iframe route on the top.
+        // In that case we don't want to change the iframe URL.
+        if (!routeName) {
+            return;
+        }
+
+        const newOldDotURL = getOldDotURL(routeName, params);
+        if (!newOldDotURL) {
+            console.error(`No old dot URL found for routeName: ${routeName} and params: ${JSON.stringify(params)}`);
+            return;
+        }
+
+        setOldDotURL(`${BASE_IFRAME_URL}/${newOldDotURL}`);
+    }, [routeName, params]);
+
+    useEffect(() => {
+        document.cookie = `authToken=${session.authToken}; domain=expensify.com.dev; path=/;secure;`;
```

Comment:
Will this work on production since the domain is hardcoded to `expensify.com.dev` here?

Reply:
cc @hayata-suenaga

Reply:
we need to change that

Adam copied the code from my WIP PR and that's why it's hardcoded there.

@adamgrzybowski, please find if there is a constant already for the domain that changes depending on the environment.

production and staging -> `expensify.com`
dev -> `expensify.com.dev`

Reply:
So I found something like this
```
// The DEFAULT API is the API used by most environments, except staging, where we use STAGING (defined below)
// The "staging toggle" in settings toggles between DEFAULT and STAGING APIs
// On both STAGING and PROD this (DEFAULT) address points to production
// On DEV it can be configured through ENV settings and can be a proxy or ngrok address (defaults to PROD)
// Usually you don't need to use this URL directly - prefer `ApiUtils.getApiRoot()
```

Does that mean that I should use `ApiUtils.getApiRoot()` ?